component/org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/mutualtls/utils/MutualTLSUtil.java

```diff
@@ -70,4 +101,89 @@ public static String hexify(byte bytes[]) {
         }
         return builder.toString();
     }
+    /**
+     * Fetch JWKS endpoint using client ID.
+     *
+     * @param clientID  client ID
+     */
+    public static URL getJWKSEndpoint(String clientID) throws OAuthClientAuthnException {
+
+        String jwksUri = StringUtils.EMPTY;
+        ServiceProviderProperty[] spProperties;
+        try {
+            ServiceProvider serviceProvider = OAuth2Util.getServiceProvider(clientID);
+            spProperties = serviceProvider
+                    .getSpProperties();
+        } catch (IdentityOAuth2Exception e) {
+            throw new OAuthClientAuthnException("Error while getting the service provider for client ID " +
+                    clientID, OAuth2ErrorCodes.SERVER_ERROR, e);
+        }
+
+        if (spProperties != null) {
+            for (ServiceProviderProperty spProperty : spProperties) {
+                if (JWKS_URI.equals(spProperty.getName())) {
+                    jwksUri = spProperty.getValue();
+                    break;
+                }
+            }
+        } else {
+            throw new OAuthClientAuthnException("Error while fetching jwks URL the URL might be malformed " +
+                    clientID, OAuth2ErrorCodes.INVALID_REQUEST);
+        }
+        URL url;
+        try {
+            url = new URL(jwksUri);
+        } catch (MalformedURLException e) {
+            throw new OAuthClientAuthnException("Error while fetching jwks URL the URL might be malformed " +
+                    clientID, OAuth2ErrorCodes.INVALID_REQUEST, e);
+        }
+        return url;
+    }
+
+    /**
+     * Fetch JWK Set as a String from JWKS endpoint.
+     *
+     * @param jwksUri JWKS Endpoint URL
+     */
+    public static String getResourceContent(URL jwksUri) throws IOException {
+
+        Resource resource = null;
+        JWKSCacheKey jwksCacheKey = new JWKSCacheKey(jwksUri.toString());
+        JWKSCacheEntry jwksCacheEntry = JWKSCache.getInstance().getValueFromCache(jwksCacheKey);
+        if (jwksCacheEntry != null) {
+            resource = jwksCacheEntry.getValue();
+            if (log.isDebugEnabled()) {
+                log.debug("Retrieving JWKS for " + jwksUri.toString() + " from cache.");
+            }
+        }
+        if (resource == null) {
+            DefaultResourceRetriever defaultResourceRetriever = new DefaultResourceRetriever();
```

Comment:
get the configs from Identity.XML connect time out and read Time out and call relevant constructor